tests/integration/testsuites/custom-domain/testsuite.go

```diff
@@ -108,15 +109,14 @@ func (t *testsuite) TearDown() {
 }
 
 func (t *testsuite) BeforeSuiteHooks() []func() error {
-	return []func() error{hooks.ApplyAndVerifyApiGatewayCrSuiteHook}
+	return []func() error{hooks.DeleteBlockingResourcesSuiteHook, hooks.ApplyAndVerifyApiGatewayCrSuiteHook}
```

Comment:
Why is this added as a before hook? Is it a safe-guard to retry the deletion, if it failed in the After hook?
I think we should fail all the tests if a hook wasn't able to successfully clear the state, because all following tests might be affected.

Reply:
It is just to make sure we start clean. Perhaps we do not need it. Will try to have it just before deleting CR so I do not block because some resources are left uncleaned. It is a leftover of making scenarios work... faster :)